mailbox/cassandra/src/main/java/org/apache/james/mailbox/cassandra/mail/task/SolveMessageInconsistenciesService.java

```diff
@@ -0,0 +1,454 @@
+            }
+
+            public ImmutableList<ComposedMessageId> getFixedInconsistencies() {
+                return fixedInconsistencies;
+            }
+
+            public ImmutableList<ComposedMessageId> getErrors() {
+                return errors;
+            }
+
+            @Override
+            public final boolean equals(Object o) {
+                if (o instanceof Snapshot) {
+                    Snapshot snapshot = (Snapshot) o;
+
+                    return Objects.equals(this.processedImapUidEntries, snapshot.processedImapUidEntries)
+                        && Objects.equals(this.processedMessageIdEntries, snapshot.processedMessageIdEntries)
+                        && Objects.equals(this.addedMessageIdEntries, snapshot.addedMessageIdEntries)
+                        && Objects.equals(this.updatedMessageIdEntries, snapshot.updatedMessageIdEntries)
+                        && Objects.equals(this.removedMessageIdEntries, snapshot.removedMessageIdEntries)
+                        && Objects.equals(this.errors, snapshot.errors)
+                        && Objects.equals(this.fixedInconsistencies, snapshot.fixedInconsistencies);
+                }
+                return false;
+            }
+
+            @Override
+            public final int hashCode() {
+                return Objects.hash(processedImapUidEntries, processedMessageIdEntries, addedMessageIdEntries, updatedMessageIdEntries, removedMessageIdEntries, fixedInconsistencies, errors);
+            }
+
+            @Override
+            public String toString() {
+                return MoreObjects.toStringHelper(this)
+                    .add("processedImapUidEntries", processedImapUidEntries)
+                    .add("processedMessageIdEntries", processedMessageIdEntries)
+                    .add("addedMessageIdEntries", addedMessageIdEntries)
+                    .add("updatedMessageIdEntries", updatedMessageIdEntries)
+                    .add("removedMessageIdEntries", removedMessageIdEntries)
+                    .add("fixedInconsistencies", fixedInconsistencies)
+                    .add("errors", errors)
+                    .toString();
+            }
+        }
+
+        private final AtomicLong processedImapUidEntries;
+        private final AtomicLong processedMessageIdEntries;
+        private final AtomicLong addedMessageIdEntries;
+        private final AtomicLong updatedMessageIdEntries;
+        private final AtomicLong removedMessageIdEntries;
+        private final ConcurrentLinkedDeque<ComposedMessageId> fixedInconsistencies;
+        private final ConcurrentLinkedDeque<ComposedMessageId> errors;
+
+        Context() {
+            this(new AtomicLong(), new AtomicLong(), new AtomicLong(), new AtomicLong(), new AtomicLong(), ImmutableList.of(), ImmutableList.of());
+        }
+
+        Context(long processedImapUidEntries, long processedMessageIdEntries, long addedMessageIdEntries, long updatedMessageIdEntries,
```

Comment:
we usually don't use implement more than one constructor and rely on factory methods if we need different argument sets.

Reply:
@LanKhuat https://github.com/chibenwa/james-project/commit/ee08353cdc70808c1712cefe9fca76c8c07b9fff should be fixing it.

Feel free to cherry-pick it.

https://github.com/chibenwa/james-project/commits/cassandra-message-inconsistencies